### PR TITLE
Separate API Auth and Anypoint Headers  (and some fixes)

### PR DESCRIPTION
**Pull request by titans-of-code**
# Features

**Separate Anypoint Headers from API Auth**
  - metrics-accelerator-api.raml: moved anypoint headers to Anypoint headers trait and removed from securedBy.  The headers are the Anypoint Platform security, NOT necessarily the security for this API.  Keeping them out of the Authorization header allows for implementers to use their own security such as OAuth.  The Anypoint security headers must still be provided in addition to any security added by the implementer.
  - Added feature to allow Base64 or plain text secrets in both secret/password Anypoint headers.

**Added properties and loggers for API and Dash**
  - Added API and Dashboard HTTP values as properties in app-dev.yaml and api.xml.
  - Added loggers to dashboard choices for debugging.  Added 501-Not Implemented if dashboard is disabled.

# Fixes
- Fixed api.xml flow names to match flows Studio generates from metrics-accelerator-api.raml.  Updated MUnit flows to match.
- Updated latest version of dependencies
- api-call-api-platform: Added default empty string to avoid null error when no response code provided.

### CODE REVIEW

**Comment by panizzag** [comment]
Reverted dependencies versions. Should be a different feature. Also, I've been facing [this](https://help.mulesoft.com/s/article/MUnit-Maven-Plugin-2-3-5-Fails-to-Find-Mule-Embedded-API-Dependency-1-3-0-20210622) issue, result of using MUnit Maven Plugin 2.3.5.

**Comment by panizzag** [comment]
image

Latest tests execution after changes

**Comment by panizzag** [comment]
Tests re ran after fix on Base64 for Connected Apps client secret
image

Connected App Login after change:
image

File: src/main/resources/api/traits/connected-app-auth.raml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    description: |
+      Receives the connected app client id used to login to Anypoint Platform.
+    type: string
+    required: false
+  X-ANYPNT-CLIENT-SECRET:
+    description: |
+      Receives a connected app client secret to login to Anypoint Platform.
+    type: string
+    required: false
+  X-ANYPNT-ORG-ID:
```
Comment (richardmckinley) [inline src/main/resources/api/traits/connected-app-auth.raml:15]: Duplicate of the header in x-anypnt-org-id.raml

Comment (panizzag) [reply src/main/resources/api/traits/connected-app-auth.raml]: Done. Thanks for pointing this out @richardmckinley 👍

**Comment by DavoCoder** [comment]
Everything looks good, great job adjusting this PR @panizzag @richardmckinley, if we are ready with this we can merge it and from develop run some last tests after the release

**Review by DavoCoder — APPROVED** [review]
Everything looks good, great job adjusting this PR @panizzag @richardmckinley, if we are ready with this we can merge it and from develop run some last tests after the release

**Review by richardmckinley — APPROVED** [review]

**Review by panizzag — COMMENTED** [review]
LGTM

**Pull request merged**